src/main/java/coloringpage/Frame.java

```diff
@@ -1,4 +1,130 @@
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class Frame extends JFrame {
+
+    private JPanel topPanel;
+    private JTextField pathField;
+
+    private JPanel middlePanel;
+    private final JLabel originalImageLabel = new JLabel();
+    private final JLabel finalImageLabel = new JLabel();
+
+    private JPanel bottomPanel;
+
+    ImageController controller = new ImageController(originalImageLabel, finalImageLabel);
+    private final  JFileChooser fileChooser = new JFileChooser();
+
+    public Frame() {
+        super();
+        setSize(800, 500);
+        setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+        setTitle("Coloring Page Maker");
+        setLayout(new BorderLayout());
+
+        setTopPanel();
+        setImagesPanel();
+        setBottomPanel();
+
+
+        add(topPanel, BorderLayout.PAGE_START);
+        add(middlePanel, BorderLayout.CENTER);
+        add(bottomPanel, BorderLayout.PAGE_END);
+    }
+
+    private void setTopPanel() {
+        topPanel = new JPanel();
+        JLabel imagePathLabel = new JLabel("Enter image url or Browse:");
+        imagePathLabel.setOpaque(true);
+
+        pathField = new JTextField(30);
+
+        JButton browseButton = new JButton("Browse");
+        browseButton.addActionListener(this::browseFiles);
+
+
+        JButton goButton = new JButton("Convert");
+        goButton.addActionListener(evt -> {
+            try {
+                convertImage(evt);
+            } catch (MalformedURLException e) {
+                e.printStackTrace();
+            }
+        });
+
+        topPanel.add(imagePathLabel);
+        topPanel.add(pathField);
+        topPanel.add(browseButton);
+        topPanel.add(goButton);
+    }
+
+    private void setImagesPanel() {
+        middlePanel = new JPanel();
+        middlePanel.setLayout(new GridLayout(1,2));
+        middlePanel.add(originalImageLabel);
+        middlePanel.add(finalImageLabel);
+    }
+
+    private void setBottomPanel() {
+        bottomPanel = new JPanel();
+        bottomPanel.setLayout(new FlowLayout());
+        JButton saveButton = new JButton("Save As");
+        saveButton.addActionListener(this::saveLineImage);
+
+        bottomPanel.add(saveButton);
+    }
+
+
+    private void browseFiles(ActionEvent evt) {
+        fileChooser.setFileFilter(new FileNameExtensionFilter("Images", "jpeg", "jpg", "png"));
+        fileChooser.setCurrentDirectory(new File(System.getProperty("user.home")));
+        fileChooser.setFileSelectionMode(JFileChooser.FILES_ONLY);
+
+        int result = fileChooser.showOpenDialog(this);
+        if (result == JFileChooser.APPROVE_OPTION) {
+            pathField.setText(fileChooser.getSelectedFile().getAbsolutePath());
+        }
+    }
+
+    private void convertImage(ActionEvent evt) throws MalformedURLException {
```

Comment:
such cool logic!

Reply:
thank you thank you
I would like to begin this speech by thanking google and stack overflow for being there in my darkest moments and solving all my problems....